plugins/topology/README.md

```diff
@@ -1,44 +1,56 @@
-4. To get the resources from a k8s cluster add the following annotations to the entity's catalog-info.yaml as K8s plugin identifies if the provided entity has k8s resources and if yes from which namespace it should get the resources based on the following annotations.
+1. Install and configure the Kubernetes plugin by following the [installation](https://backstage.io/docs/features/kubernetes/installation) and [configuration](https://backstage.io/docs/features/kubernetes/configuration) guides.
+2. The Kubernetes plugin is configured and able to connect to the cluster using a `ServiceAccount`.
+3. The [`ClusterRole`](https://backstage.io/docs/features/kubernetes/configuration#role-based-access-control) must be granted to `ServiceAccount` accessing the cluster. If you have the Backstage Kubernetes plugin configured, then the `ClusterRole` is already granted.
+4. To receive the resources from a Kubernetes cluster, add the following annotations to the entity's `catalog-info.yaml` file:
+
    ```yaml
    annotations:
      backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>
      backstage.io/kubernetes-namespace: <RESOURCE_NS>
    ```
-5. A custom label selector can also be added which will then be used by Backstage to find the resources. The label selector takes precedence over the id annotation.
+
+   The Kubernetes plugin identifies if the provided entity contains Kubernetes resources and from which namespace the plugin receives the resources based on the previous annotations.
+
+5. You can also add a custom label selector, which Backstage uses to find the resources. The label selector takes precedence over the ID annotation.
    `'backstage.io/kubernetes-label-selector': 'app=my-app,component=front-end`
-6. For k8s plugin to get the desired entity's k8s resources the following label should be added to the resources(if label selector is used then the labels mentioned in that should be present on the resource):
+6. You must add the following label to the resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:
+
    ```yaml
    'backstage.io/kubernetes-id': <BACKSTAGE_ENTITY_NAME>`
    ```
-7. Workload resources like Deployments and Pods will be shown in a visual group based on below label
+
+   ***
+
+   **NOTE**
+
+   When using the label selector, the mentioned labels must be present on the resource.
+
+   ***
+
+7. The workload resources such as Deployments and Pods are displayed in a visual group based on the following label:
    ```yaml
    'app.kubernetes.io/part-of': <GROUP_NAME>
    ```
-8. Workload resources like Deployment or Pod will be shown with a visual connector based on below annotations
+8. The workload resources such as Deployment or Pod are displayed with a visual connector based on the following annotations:
    ```yaml
    'app.openshift.io/connects-to': '[{"apiVersion": <RESOURCE_APIVERSION>,"kind": <RESOURCE_KIND>,"name": <RESOURCE_NAME>}]'
    ```
```

Comment:
On the OpenShift console documentation we do have links to the labeling guidelines. Should we do something similiar here?

Reply:
@gorkem Could you please share the reference?

Reply:
ON 4.12 docs [this section](https://docs.openshift.com/container-platform/4.12/applications/odc-viewing-application-composition-using-topology-view.html#odc-labels-and-annotations-used-for-topology-view_viewing-application-composition-using-topology-view) explains the use of labels

Reply:
@gorkem thanks for sharing the reference. 
@invincibleJai Could you please see [this section](https://docs.openshift.com/container-platform/4.12/applications/odc-viewing-application-composition-using-topology-view.html#odc-labels-and-annotations-used-for-topology-view_viewing-application-composition-using-topology-view) and let me know if we can add it to our README?

Reply:
@hmanwani-rh in the reference shared is OCP docs so am not sure if we need to add that here,  not sure if  @gorkem  meant that but OCP docs refer to https://github.com/redhat-developer/app-labels/blob/master/labels-annotation-for-openshift.adoc so maybe we can do something similar here

>For detailed information on the labels and annotations OpenShift Container Platform applications must use, see [Guidelines for labels and annotations for OpenShift applications](https://github.com/redhat-developer/app-labels/blob/master/labels-annotation-for-openshift.adoc).

Reply:
Yes, I think just referencing the link to "Guidelines for labels and annotations"  should be enough.

Reply:
Done